reporter/internal/pdata/generate.go

```diff
@@ -158,44 +163,36 @@ func (p *Pdata) setProfile(
 				address:   uint64(frame.AddressOrLineno),
 				frameType: frame.Type.String(),
 			}
-			switch frameKind := frame.Type; frameKind {
-			case libpf.NativeFrame:
-				// As native frames are resolved in the backend, we use Mapping to
-				// report these frames.
-				locationMappingIndex, exists := mappingSet.AddWithCheck(frame.FileID)
-				if !exists {
-					ei, exists := p.Executables.GetAndRefresh(frame.FileID,
-						ExecutableCacheLifetime)
-					// Next step: Select a proper default value,
-					// if the name of the executable is not known yet.
-					fileName := "UNKNOWN"
-					if exists {
-						fileName = ei.FileName
-					}
 
+			if frame.MappingFile != (libpf.FrameMappingFile{}) {
+				index, ok := mappingSet.AddWithCheck(uniqueMapping{
```

Comment:
There's a semantic change from the previous logic here, we're now generating mappings for frames with 'kernel' frame type. All such mappings will have '0' as start address, but the filename will either be 'vmlinux' or (I'm guessing) the kernel module. I think this is fine as it gives us more information.

Reply:
Correct, this was also intended change. Sorry for not pointing out. I assumed reporting the "backing" file would not violate anything but did not check specs.